promise.go

```diff
@@ -56,6 +58,14 @@ type Promise struct {
 	wg *sync.WaitGroup
 }
 
+// Result is used by methods that need to return result
+// 	objects (namely `AllSettled` method)
+type Result struct {
```

Comment:
Can we keep the unneeded commits out of this PR and focus only on the Race method. We push this into another PR for the AllSettled method

Reply:
Sure, (:-), just saw this jehjeh, will make the change :+1: ),